### ISSUE TITLE
How does the training work?

### ISSUE THREAD

**Issue by stephenllh** [wzhshi/SCSNet#3]
I want to ask a question to clarify certain training details. In the paper, you mentioned that "Suppose the reconstruction network has T initial reconstructions and T final reconstructions, we have 2T objectives to minimize." Does that mean the "initial reconstruction network" and "deep reconstruction network" are trained with MSE loss separately, although the input of "deep" is the output of "initial"?

**Comment by wzhshi** [comment]
They are trained jointly.  Please see my codes for more details.

2021-03-29 18:20:39 "Stephen Lau" ***@***.***> 写道：

I want to ask a question to clarify certain training details. In the paper, you mentioned that "Suppose the reconstruction network has T initial reconstructions and T final reconstructions, we have 2T objectives to minimize." Does that mean the "initial reconstruction network" and "deep reconstruction network" are trained with MSE loss separately, although the input of "deep" is the output of "initial"?

**Comment by stephenllh** [comment]
Then does the gradient of the "deep reconstruction network" flow to the "initial reconstruction block"?

**Comment by wzhshi** [comment]
Yes. It accumulates the gradient of the deep network and the gradient of the loss function of the initial reconstruction network when optimizing the sampling network and the initial reconstruction network.​


2021-03-30 15:32:19 "Stephen Lau" ***@***.***> 写道：

Then does the gradient of the "deep" flow to the "initial"?

**Comment by stephenllh** [comment]
Thank you for taking the time to reply. It is very helpful.
